### PR TITLE
Rework Hasher

### PR DESCRIPTION
**Pull request by krienow**
This PR suggests multiple changes:

- Set algorithm to `PASSWORD_DEFAULT`. This supports stronger algorithms in the future.
- Rename BcryptHasher to Hasher accordingly.
- Simplify code.

### CODE REVIEW

File: src/Illuminate/Hashing/Hasher.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class Hasher implements HasherContract
+{
+    /**
+     * Hash the given value.
+     *
+     * @param  string  $value
+     * @param  array   $options
+     * @return string
+     *
+     * @throws \RuntimeException
```
Comment (m1guelpf) [inline src/Illuminate/Hashing/Hasher.php]: When does it throw a RuntimeException?

Comment (krienow) [inline src/Illuminate/Hashing/Hasher.php]: No, I overlooked this. 😄

**Comment by taylorotwell** [comment]
No plans on changing this right now.

**Comment by krienow** [comment]
Why not? This would allow Laravel to support better password hashing algorithms by default. Existing passwords could be easily rehashed.

**Pull request closed without merging**